src/renderers/webgl/WebGLPrograms.js

```diff
@@ -97,6 +97,12 @@ function WebGLPrograms( renderer, cubemaps, cubeuvmaps, extensions, capabilities
 
 		}
 
+		if ( map && map.isCompressedTexture ) {
```

Comment:
Just for clarification why there is no WebGL 2 check:

The problem is that `WEBGL_compressed_texture_s3tc_srgb` has to be requested in `WebGLUtils` and this extension does exist for WebGL 1. If we add a check for WebGL 2, WebGL 1 devices will apply the sRGB decode twice. So we have to make this step and disable the inline decode for all compressed textures.

The entire sRGB texture handling in the engine is tricky right now. But it will become more straightforward if we remove the inline decode.  This will allow us to completely remove stuff like `getTextureEncodingFromMap()` and also this code.

Reply:
We might still need `getTextureEncodingFromMap` if we want to support wide-gamut linear rendering pipelines in WebGL2 someday. But I'd also be OK with just saying that feature would (if added) only be available in WebGPU. Related WebGPU issue: https://github.com/gpuweb/gpuweb/issues/1715